tests/client/api_v1/test_get_balance.c

```diff
@@ -0,0 +1,81 @@
+// Copyright 2020 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unity/unity.h>
+
+#include "client/api/v1/get_balance.h"
+#include "core/utils/byte_buffer.h"
+
+#define ADDR_HEX "7ed3d67fc7b619e72e588f51fef2379e43e6e9a856635843b3f29aa3a3f1f006"
+
+void test_get_balance() {
+  iota_client_conf_t conf = {
+      .url = "http://0.0.0.0/",
+      .port = 14265  // use default port number
+  };
+
+  res_balance_t* res = calloc(1, sizeof(res_balance_t));
+
+  // test null cases
+  TEST_ASSERT_EQUAL_INT(-1, get_balance(NULL, NULL, NULL));
+  TEST_ASSERT_EQUAL_INT(-1, get_balance(NULL, NULL, res));
+  TEST_ASSERT_EQUAL_INT(-1, get_balance(&conf, NULL, NULL));
+  TEST_ASSERT_EQUAL_INT(-1, get_balance(&conf, NULL, res));
+
+  // test for success
+  TEST_ASSERT_EQUAL_INT(0, get_balance(&conf, ADDR_HEX, res));
+}
+
+void test_deser_balance_info() {
+  char const* json_info_200 =
+      "{\"data\":{\"address\": \"7ed3d67fc7b619e72e588f51fef2379e43e6e9a856635843b3f29aa3a3f1f006\","
+      "\"maxResults\": 1000,"
+      "\"count\": 25,"
+      "\"balance\": 1338263}}";
+
+  char const* json_info_400 =
+      "{\"error\": {"
+      "\"code\": \"invalid_data\", "
+      "\"message\": \"invalid data provided\"}}";
+
+  char const* json_info_404 =
+      "{\"error\": {"
+      "\"code\": \"not_found\", "
+      "\"message\": \"could not find data\"}}";
+
+  // test http status code 200
+  res_balance_t* res = res_balance_new();
+  TEST_ASSERT_EQUAL_INT(0, deser_balance_info(json_info_200, res));
+  TEST_ASSERT_EQUAL_INT(1000, res->u.output_balance->max_results);
+  TEST_ASSERT_EQUAL_INT(25, res->u.output_balance->count);
+  TEST_ASSERT_EQUAL_INT(1338263, res->u.output_balance->balance);
+
+  // test http status code 400
+  TEST_ASSERT_EQUAL_INT(-1, deser_balance_info(json_info_400, res));
+  TEST_ASSERT(res->is_error);
+  TEST_ASSERT_EQUAL_STRING("invalid_data", res->u.error->code);
+  TEST_ASSERT_EQUAL_STRING("invalid data provided", res->u.error->msg);
+
+  // reset res->is_error to false;
+  res->is_error;
+
+  // test http status code 404
+  TEST_ASSERT_EQUAL_INT(-1, deser_balance_info(json_info_404, res));
+  TEST_ASSERT(res->is_error);
+  TEST_ASSERT_EQUAL_STRING("not_found", res->u.error->code);
+  TEST_ASSERT_EQUAL_STRING("could not find data", res->u.error->msg);
+
+  res_balance_free(res);
+}
+
+int main() {
+  UNITY_BEGIN();
+
+  RUN_TEST(test_deser_balance_info);
+  RUN_TEST(test_get_balance);
```

Comment:
please make sure you pass this case in local. CI will not pass because of the `0.0.0.0` endpoint. 
```suggestion
  // RUN_TEST(test_get_balance);
```

Reply:
solved via 130560a9f9b7de6a0d344bc418717d7aff63dbbe